Assets/MRTK/SDK/Editor/Inspectors/UX/StateVisualizer/StateVisualizerInspector.cs

```diff
@@ -0,0 +1,402 @@
+                                }
+
+                                animationTargetList.DeleteArrayElementAtIndex(j);
+                                break;
+                            }
+                        }
+
+                        using (new EditorGUILayout.VerticalScope())
+                        {
+                            if (targetObj.objectReferenceValue != null)
+                            {
+                                InspectorUIUtility.DrawDivider();
+
+                                GameObject targetGameObject = targetObj.objectReferenceValue as GameObject;
+
+                                // Ensure the target game object is valid
+                                if (IsTargetObjectValid(targetGameObject))
+                                {
+                                    string animatablePropertiesFoldoutID = stateContainerName.stringValue + "AnimatableProperties" + "_" + targetGameObject.name + target.name;
+
+                                    if (InspectorUIUtility.DrawSectionFoldoutWithKey(targetGameObject.name + " Animatable Properties", animatablePropertiesFoldoutID, MixedRealityStylesUtility.BoldFoldoutStyle, false))
+                                    {
+                                        using (new EditorGUI.IndentLevelScope())
+                                        {
+                                            RenderAnimatablePropertyList(animatablePropertyList, stateContainerName, j);
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    targetObj.objectReferenceValue = null;
+                                    Debug.LogError("The target object must be itself or a child object");
+                                }
+                            }
+
+                            EditorGUILayout.Space();
+                        }
+                    }
+                }
+
+                EditorGUILayout.Space();
+
+                RenderAddTargetButton(animationTargetList);
+            }
+        }
+
+        private void RenderAddTargetButton(SerializedProperty animationTargetList)
+        {
+            if (GUILayout.Button("Add Target"))
+            {
+                animationTargetList.InsertArrayElementAtIndex(animationTargetList.arraySize);
+
+                serializedObject.ApplyModifiedProperties();
+
+                SerializedProperty newAnimationTarget = animationTargetList.GetArrayElementAtIndex(animationTargetList.arraySize - 1);
+                newAnimationTarget.FindPropertyRelative("target").objectReferenceValue = null;
+
+                SerializedProperty stateAnimatablePropertiesList = newAnimationTarget.FindPropertyRelative("stateAnimatableProperties");
+
+                // Clear the new list
+                for (int k = 0; k < stateAnimatablePropertiesList.arraySize; k++)
```

Comment:
Could this be why you were seeing that null reference error that you ran into when deleting a property? Try iterating through the list backwards, or see if stateAnimatablePropertiesList has a method to clear the array.

If that doesn't fix it, treat this comment as a reminder to fix that null reference error you hit when trying to remove a property from an animation that had just one element in it :)

Reply:
I fixed this issue, another index value should have been the parameter for removing the keyframes